packages/flutter/lib/src/cupertino/magnifier.dart

```diff
@@ -245,29 +245,60 @@ class CupertinoMagnifier extends StatelessWidget {
         color: Color.fromARGB(25, 0, 0, 0),
         blurRadius: 11,
         spreadRadius: 0.2,
+        blurStyle: BlurStyle.outer,
       ),
     ],
+    this.clipBehavior = Clip.none,
     this.borderSide =
         const BorderSide(color: Color.fromARGB(255, 232, 232, 232)),
     this.inOutAnimation,
   });
 
-  /// The shadows displayed under the magnifier.
+  /// A list of shadows cast by the [Magnifier].
+  ///
+  /// If the shadows use a [BlurStyle] that paints inside the shape, or if they
+  /// are offset, then a [clipBehavior] that enables clipping (such as
+  /// [Clip.hardEdge]) is recommended, otherwise the shadow will occlude the
+  /// magnifier (the shadow is drawn above the magnifier so as to not be
+  /// included in the magnified image).
```

Comment:
Thanks for explaining this stuff in the docs, I remember it being very tricky when implemented.